accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,298 @@
+
+  #### Cluster node requirements
+
+  Every machine that needs to be part of cluster (either a master or a worker) has strictly hardware and software requirements (e.g.: every machine has to run SUSE Linux Enterprise 15, has a fixed IP that can be resolved, [a precise partition layout with no swap](https://susedoc.github.io/doc-caasp/beta/caasp-deployment/single-html/#_suse_linux_enterprise_server_installation), at least 2 CPUs, etc.). From now on, these requirements will be called "CaaS Platform node requirements".
+
+  For this reason, the CaaS Platform [supports deployment on](https://susedoc.github.io/doc-caasp/beta/caasp-deployment/single-html/#_platform):
+
+  - VMware ESXi
+  - SUSE OpenStack Cloud 8
+  - [Bare-metal deployment](https://susedoc.github.io/doc-caasp/beta/caasp-deployment/single-html/#_deployment_on_bare_metal)
+
+  CaaS Platform also provides templates for the above providers that can be found in the `skuba` package (in `/usr/share/caasp/{AutoYaST, terraform}`). For VMware and SUSE OpenStack Cloud, the template is provided via Terraform. The up-to-date version of these templates can be found at:
+  - for VMware: https://github.com/SUSE/skuba/tree/master/ci/infra/vmware
+  - for SUSE OpenStack Cloud 8: https://github.com/SUSE/skuba/tree/master/ci/infra/openstack
+  - for AutoYaST: https://github.com/SUSE/skuba/blob/master/ci/infra/bare-metal/AutoYaST.xml
+
+  One additional scenario would be to use [an existing node running SUSE Linux Enterprise 15](https://susedoc.github.io/doc-caasp/beta/caasp-deployment/single-html/#_deployment_on_existing_sles_installation). In this particular case, we should check that the CaaS Platform requirements are satisfied before offering this node to be part of the cluster.
+  These checks can be done using Salt grains. 
+
+  ##### Templating selection
+  
+  SUSE Manager has an existing integration with VMWare (Virtual Host Manager). This integration is read-only, as SUSE Manager can show the guests running in the VMWare host and display information about the guests.
+  Currently, there is no any integration with SUSE OpenStack Cloud 8.
+  Additionally, SUSE Manager has not currently any integration with `terraform`.
+
+  SUSE Manager supports autoinstallation via AutoYaST.
+
+  For the reasons above, it looks promising to invest in two directions for fulfilling the integration in the deployment phase:
+  1. Existing node: SUSE Manager must filter the eligible nodes that satisfy all CaaS Platform node requirements among the already registered Salt minions already available
+  2. AutoYaST deployment: in case the user wants to deploy additional autoinstalled node, SUSE Manager should guide the user into setting up as many nodes of the cluster by autoinstalling them using the AutoYaST profile provided by `skuba`.
+
+  **NOTE**: skuba also provides templates for AWS and libvirt, but those are not supported at the moment of writing.
+
+  ##### The Management Node
```

Comment:
During the lifecycle of the cluster, this "management" node is not required. I'd rather think of it as a "bootstrap" machine that can live outside of the final cluster (e.g. a laptop or machine that has access to the target machines to be initialized as a Kubernetes cluster).

We generate the so-called "cluster definition folder", but to keep it in mind: this state in this folder should be completely optional in the short term (it's not the situation at the moment, mainly because of the control plane and worker templates). However, and this is what we would like to achieve soon, is that all state of the cluster resides on the cluster, so once bootstrapped there's no real need to have state somewhere else, any person on any machine that has a `kubeconfig` file with enough permissions could manage the cluster lifecycle using state present exclusively inside the cluster.

Reply:
So does that mean a management node can be re-used to fire up any clusters?

In that case in SUMA, we can model management nodes just like build hosts building images; which will run skuba commands on any cluster.

Reply:
> So does that mean a management node can be re-used to fire up any clusters?

Absolutely! They just need different "cluster definition folders" (regular filesystem folder). In the near future nothing at all, since all state we need will reside within the cluster itself (you'd only need a way to authenticate against the cluster, e.g. a `kubeconfig` file with some special rights to perform certain management actions on the cluster).

Reply:
In this RFC, the `Management Node` was described as being a special citizen in SUSE Manager (assimilable to build hosts for containers). The idea was to keep the Management Node around to deploy all the clusters (not only one). It was also intended that the `Management Node` must keep the `definition folder` and the `kubeconfig` in the future.
Does this clarify what was the intention?